tests/integ/modin/index/test_unique.py

```diff
@@ -33,7 +33,36 @@ def test_index_unique_nan_vs_none_negative():
 
 
 @pytest.mark.parametrize("native_index", NATIVE_INDEX_UNIQUE_TEST_DATA)
+@pytest.mark.parametrize("dropna", [True, False])
 @sql_count_checker(query_count=1)
-def test_index_nunique(native_index):
+def test_index_nunique(native_index, dropna):
     snow_index = pd.Index(native_index)
-    assert snow_index.nunique() == native_index.nunique()
+    assert snow_index.nunique(dropna=dropna) == native_index.nunique(dropna=dropna)
+
+
+def test_index_unique_data_columns_should_not_affect_index_column():
+    native_df = native_pd.DataFrame(
+        {
+            "B": [10, 20, 30, 40, 50],
+            "A": [11, 22, 33, 44, 55],
+            "C": [60, 70, 80, 90, 100],
+        },
+        index=native_pd.Index([5, 4, 3, 2, 1], name="A"),
+    )
+    snow_df = pd.DataFrame(native_df)
+
+    with pytest.raises(AssertionError):
+        # Assert that even though the index column and a data column have
+        # the same name "A", the correct column is picked for both cases and
+        # different results are produced.
+        snow_index = snow_df.index
+        snow_df_a = snow_df.A
+        with SqlCounter(query_count=1):
+            assert_index_equal(snow_index.unique(), snow_df_a.unique())
+
+    with SqlCounter(query_count=1):
+        assert_index_equal(snow_df.index.unique(), native_df.index.unique())
+    # TODO: SNOW-1524901: snow_df.A.unique() does not produce the correct result here.
```

Comment:
this should be a separate test case in `series/test_unique.py` marked as `pytest.mark.xfail(strict=True)`